internal/controllers/machinedeployment/mdutil/util.go

```diff
@@ -432,6 +432,7 @@ func MachineTemplateDeepCopyRolloutFields(template *clusterv1.MachineTemplateSpe
 	templateCopy.Annotations = nil
 
 	// Drop node timeout values
+	templateCopy.Spec.MinReadySeconds = nil
```

Comment:
Please add test coverage. Should be TestMachineTemplateUpToDate => machineTemplateWithDifferentInPlaceMutableSpecFields